### DISCUSSION TITLE
How do TileSource min/max zoom parameters affect displayed map

### DISCUSSION THREAD

**Discussion by demiantres** [mapsforge/vtm#896] [Q&A] [answered]
Hi,

The TileSource offers min/max/over zoom parameters in the constructor. The TileManager later gets these levels using TileSource.getZoomLevelMin() (...). 

What do this zoom levels exactly do? I tried to set the min/max zoom levels for different map sources and they seem to have no (or hardly any) effect on the displayed map during zooming.

My initial idea was to use these levels for limiting the map. For instance when the min-max levels are set to 5 and 10 then I expected that the map is only shown when the zoom level is between these levels?

Regards,

**Comment by devemux86** [comment]
Map limits can be defined via the various setters in [Viewport](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/map/Viewport.java).

**Reply by demiantres** [reply]
Let's say there are two maps, each loaded in a different layer:
Layer 1: An overview map for levels 0-5.
Layer 2: A detailed map for levels 5-10.

Do I understand it correctly that setting the min/max zoom levels for each tile source will not affect when the map is displayed but instead I have to implement my own logic for enabling/disabling the layers?

**Comment by devemux86** [comment, marked as answer]
Tile source zooms indicate the data limits.
e.g. bitmap tiles <18, vector tiles <14, etc.

Map engine's limits can be set via [Viewport](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/map/Viewport.java).

**Reply by demiantres** [reply]
And if a tile source does not have data for a specific zoom level, then parent level's data is used?

**Comment by devemux86** [comment]
This can work for vector maps / tiles.
e.g. [MapilionMvtTileSource](https://github.com/mapsforge/vtm/blob/master/vtm-mvt/src/org/oscim/tiling/source/mvt/MapilionMvtTileSource.java)

**Reply by demiantres** [reply]
Ok, thanks. I figured it out.
